### ISSUE TITLE
cattle-node-agent-windows for Windows 2004 fails to come up using k8s 1.20

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#31275]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Deploy a windows cluster - VXLAN - 1 etcd, 1 control and 1 worker node linux nodes and 3 windows worker nodes `Windows_Server-2004-English-Core-Base-2020.11.11`
- The cluster comes up Active but the cattle-node-agent-windows for Windows is in error
- Error on the pods

```
(combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "cattle-node-agent-windows-mrhhq": Error response from daemon: hcsshim::CreateComputeSystem b5dae7fba2e1c90728066832232fdba8ec8acfc2ed1f9c99ac147bf1451ffe5c: The container operating system does not match the host operating system. (extra info: {"SystemType":"Container","Name":"b5dae7fba2e1c90728066832232fdba8ec8acfc2ed1f9c99ac147bf1451ffe5c","Owner":"docker","VolumePath":"\\\\?\\Volume{6a8c6a78-4ac8-4c06-9894-7e883e39a36b}","IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\docker\\windowsfilter\\b5dae7fba2e1c90728066832232fdba8ec8acfc2ed1f9c99ac147bf1451ffe5c","Layers":[{"ID":"45a5bd79-1cc8-5a3d-9e39-e2ac6550ee8e","Path":"C:\\ProgramData\\docker\\windowsfilter\\fac1938e518de8289a33f3ef1281cd96ad13b6609edfda4a0aea394fe40df58e"},{"ID":"531ef1be-a28c-5358-8476-80386b25d592","Path":"C:\\ProgramData\\docker\\windowsfilter\\b8d1ef66b624bb2a673d268961b69d2681a189c1014b3e9e205d689d66a6216f"},{"ID":"db7307f7-7aba-593c-a9e1-a6182303aa7f","Path":"C:\\ProgramData\\docker\\windowsfilter\\d3fd7fcec9821204103a5d504e19b66a98705215e24105c4405ce95515a60cec"},{"ID":"e7ae8a78-ecbf-5917-8768-78ee4c5d8d71","Path":"C:\\ProgramData\\docker\\windowsfilter\\8c2b605c36f92cd2b64441eb427ddb8bd712eac89258f5e5facafd87b4e999c5"},{"ID":"a53bd928-1f76-574c-9b21-05a220bd07ef","Path":"C:\\ProgramData\\docker\\windowsfilter\\8311133b06e4b7bc1d3b9feecf2b990d1bb333accadf66a0067890fcd49ba081"},{"ID":"ad6aa6de-8d60-5505-9602-c54a7d2853ff","Path":"C:\\ProgramData\\docker\\windowsfilter\\25d7b84ec043ae5950f42728d4ca62b301ee0d1631fd61ce4343b47fb7f8268c"},{"ID":"d4b23f67-fcf1-5aa7-868e-0bcf41d4aea3","Path":"C:\\ProgramData\\docker\\windowsfilter\\145b728cc811f282ba1d8b8a29714c5bcb93a53a4453729317ac24399aff82a3"},{"ID":"68288441-214e-5a2e-a9ad-22ce849af8b5","Path":"C:\\ProgramData\\docker\\windowsfilter\\c15bdebf8719b6fb68c181f20fb882a0de2c5edcf988797e80accccfb4f278be"},{"ID":"c78fcf72-309e-5bc6-991e-cf930d8ee8a9","Path":"C:\\ProgramData\\docker\\windowsfilter\\99ad734683873ee5191088799be69953bdaf550c110ab962b838d0459865eda7"},{"ID":"815abbfc-08bd-5a42-bbbc-52bd1a4962f7","Path":"C:\\ProgramData\\docker\\windowsfilter\\9f5d9316fc33692e9ce3957d37a823e6f14f830637e144fd94a667bc8886791b"},{"ID":"9b1a1da1-f279-5bc5-98c9-e46854c3d5ff","Path":"C:\\ProgramData\\docker\\windowsfilter\\1513df4fdf5a7fb1f5be2947d00f54e35d070c8a6ff1bd9fe2d68712ba53e458"},{"ID":"75a86828-025f-50df-8452-0104079a1bd9","Path":"C:\\ProgramData\\docker\\windowsfilter\\443ddff556a906fda97bd78a6be1cbee23ca079dd62ed1448d88eb669f3fac5a"},{"ID":"a74daa45-26a5-5780-9cc7-eab9fc39ca3b","Path":"C:\\ProgramData\\docker\\windowsfilter\\a3457fa1f0260255fb0c08e937d136c9b17a80f4487144fd9513d2c54a85421d"},{"ID":"9cfb73bd-05ef-5b20-aaac-25ee693d2bf5","Path":"C:\\ProgramData\\docker\\windowsfilter\\18160338e0519d48da69b49eb8c66e6f72762c3180d760677cdd7f854d99199f"},{"ID":"08c2b2dc-33c2-5e2a-9845-8441b151c7cb","Path":"C:\\ProgramData\\docker\\windowsfilter\\f5bbcb1cba4da159abc9ea0b85c0e4fa8ae9df1e58e338667fa6ac15e17768db"},{"ID":"25eddc50-a0ce-5c0c-8101-5e6688599d46","Path":"C:\\ProgramData\\docker\\windowsfilter\\c4b776177e101a7f84c98cb7821ed0179ee5492fcc22bc957ff99c7e85a0afdd"},{"ID":"e2d70e87-d3cd-5bb5-9e7a-c064b2690f0f","Path":"C:\\ProgramData\\docker\\windowsfilter\\7d9bfb8f99f82a8895de46b04addb3f2c11443d892460047c928f7fb90b1ab9b"},{"ID":"d4e6e076-cc6f-56d2-ae20-1dd4bec0f641","Path":"C:\\ProgramData\\docker\\windowsfilter\\437ca97fe99dea8212242d39a7f3050034ce5c2abb8c8dd96b4993b6ac13a0a3"}],"ProcessorWeight":2,"HostName":"cattle-node-agent-windows-mrhhq","HvPartition":false})
```

**Note:**
This does not happen on clusters with 1.19.7 k8s version

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5-head commit id: `761c805`
- Installation option (single install/HA): single node

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): custom windows
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
1.20.2
```

- Docker version (use `docker version`):

```
Linux nodes have 20.10.3 and windows have 19.3.13
```

**Comment by luthermonson** [comment]
@sowmyav27 this will be fixed in 1.20.3 per this pr and is what will ship in v2.5.6 https://github.com/rancher/kontainer-driver-metadata/pull/490

**Comment by sowmyav27** [comment]
**On 2.5-head commit id: `a85f4d1f8`**

- Deploy a windows cluster - VXLAN - 1 linux node all roles and 2 windows worker nodes
- On windows worker node `rancher/kubelet-pause v0.1.6 ` image is seen
- Workload is deployed in the windows worker nodes.
- Windows node agent comes up Active. No issues seen
